src/components/Badge/Badge.tsx

```diff
@@ -0,0 +1,92 @@
+import classNames from "classnames";
+import React, { HTMLProps } from "react";
+import type { ClassName, PropsWithSpread, ValueOf } from "types";
+
+export const BadgeType = {
+  ROUNDED_LARGE_NUMBER: "ROUNDED_LARGE_NUMBER",
+  UNDEFINED_LARGE_NUMBER: "UNDEFINED_LARGE_NUMBER",
+} as const;
+
+/**
+ * The props for the Badge component.
+ */
+export type Props = PropsWithSpread<
+  {
+    /**
+     * Numeric value to be displayed.
+     */
+    value: number;
+    /**
+     * The type of the badge component.
+     */
+    badgeType?: ValueOf<typeof BadgeType>;
+    /**
+     * The appearance of the badge.
+     */
+    isNegative?: boolean;
+    /**
+     * Optional class(es) to give to the badge.
+     */
+    className?: ClassName;
+  },
+  HTMLProps<HTMLSpanElement>
+>;
+
+const MAX_VAL = 999;
+const units = ["", "k", "M", "B", "T"];
+
+const round = (value: number, unit = 0) => {
+  if (value < 1000) {
+    const truncatedValue = Number(value.toString().slice(0, 4));
+    return `${truncatedValue}${units[unit]}`;
+  }
+
+  if (unit >= units.length - 1) {
+    return "+999T";
```

Comment:
This is more than 4 characters technically. And why + in the beginning?

I guess we are fine with leaving it as `999T` at that point?